openqaoa/devices.py

```diff
@@ -327,13 +329,15 @@ def check_connection(self) -> bool:
         
         # Only QPUs that are available for the specified aws region on Braket 
         # will be shown. We filter out QPUs that do not work with the circuit model
+        sesh_devices = self.aws_session.search_devices()
+        
         device_filter = np.multiply(
```

Comment:
Why are we using `np.multiply` for strings?

Reply:
From what I understand in the logic, I think it makes sure that out of the devices that are available, which ones are actually `ONLINE`. Therefore, the filtered devices only contain devices that are both available to the user && online.
@shahidee44 , can you confirm this?

Reply:
That's ok, but why are we using NumPy to do it?

Reply:
I believe its the faster filtering technique i found on stackoverflow to filter dictionaries that looked like this.
Basically both lines return a list containing booleans of whether the dictionary passed the check, if they did, we multiply both list of booleans to obtain a final list where the value of the entry is 1 iff both conditions were met.

Reply:
OK, we can leave it like this